WIP Read binary object takes zero arguments

jhammer opened the pull request:
readBinaryObject does not take an argument.

btry commented:
Hi @jhammer 

I'm the new maintainer of CFPropertyList. I refactored the code, and included your first commit of this PR.

I know you created it long ago, but do you remember why you dropped so many code in the second commit of this PR ? By reading it it seems you completely changed the way to compute the size of a binary property list, making obsolete several properties of the class. Am I right ?

jhammer commented:
Hi @btry, glad to see that CFPropertyList is getting some love. It has been a very long time, so my memory isn't clear, but I believe you are correct:

1. The problem that the second commit fixes is that the original implementation of CFPropertyList did not correctly write binary plists that contained strings with multi-byte emoji characters.

2. In the course of fixing that bug, I realized that the way `$table_offset` was being calculated was incorrect and unnecessarily complex. I simplified how `$table_offset` was calculated and removed the unnecessary properties and code.

If it helps, this code has been in production since Dec 2014 without any issues.

btry commented:
Hi

Thank you for your feedback. As you fixed something in this 2nd commit, then there is a unit test to write to make evidence of the bug, then it must pass with your fix. I'll investigate soon.

ajsb85 commented:
Hi, @jhammer 
Because this PR came from your fork 😸 
@btry will take care about that commit, no problem

Regards.
